### PR TITLE
[WR-316] add PAM Jira Plugin resource

### PR DESCRIPTION
**Pull request by pirtleshell**
adds a resource for managing PAM Jira Plugins ✨ 

tested the create, update, & destroy resource actions.

depends on un-deployed feature-branch apis & client for PAMJiraPlugin CRUD

### CODE REVIEW

File: docs/resources/tozny_pam_jira_plugin.md [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+      # Pin Tozny provider version
+      version = ">=0.14.0"
+    }
+  }
+}
+
+# Include the Tozny Terraform provider
+provider "tozny" {
+  api_endpoint     = "http://platform.local.tozny.com:8000"
+  account_username = "test+${random_string.account_username_salt.result}@tozny.com"
```
Comment (galxy25) [inline docs/resources/tozny_pam_jira_plugin.md:34]: this should follow the pattern in the other documentation (now updated to use the test-emails-group prefix)

File: docs/resources/tozny_pam_jira_plugin.md [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+# communication between TozID and Jira.
+###
+
+# Never commit your Jira API Key to source control.
+# To avoid this we create three variables pulled from the environment.
+# The names of the env variables should be prefixed with "TF_VAR_" when set in the shell.
+# If not set, terraform will prompt you for the values.
+
+# The Jira Host URL
+# export TF_VAR_JIRA_HOST_URL="tozid.atlassian.net"
```
Comment (galxy25) [inline docs/resources/tozny_pam_jira_plugin.md:90]: this `TF_VAR` is a new concept for our provider, so it needs rationale, and if the rationale is valid we should probably update the other use of environment variables(in the account resource for username and password) to also use this prefix

File: docs/resources/tozny_pam_jira_plugin.md (hunk 1 above)
Comment (pirtleshell) [reply docs/resources/tozny_pam_jira_plugin.md:34]: ah, my bad. those other documentation changes didn't exist when i copied this code 😅 will update

File: examples/realms/pamPlugins/main.tf [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+      # Pin Tozny provider version
+      version = ">=0.14.0"
+    }
+  }
+}
+
+# Include the Tozny Terraform provider
+provider "tozny" {
+  api_endpoint     = "http://platform.local.tozny.com:8000"
+  account_username = "test+${random_string.account_username_salt.result}@tozny.com"
```
Comment (galxy25) [inline examples/realms/pamPlugins/main.tf:25]: use the `test-emails-group` prefix please

File: docs/resources/tozny_pam_jira_plugin.md (hunk 2 above)
Comment (pirtleshell) [reply docs/resources/tozny_pam_jira_plugin.md:90]: it's just a suggestion. they're just terraform variables. i am happy to exclude these comments they just seemed helpful. if the env variables aren't found, terraform prompts you for the values

**Review by galxy25 — APPROVED** [review]

File: docs/resources/tozny_pam_jira_plugin.md (hunk 2 above)
Comment (galxy25) [reply docs/resources/tozny_pam_jira_plugin.md:90]: After reading https://www.terraform.io/docs/language/values/variables.html#environment-variables and your comment more I get it. I was thinking that our tozny terraform resource for the jira plugin was manually looking for these variables, and so having it look for things like `TF_VAR_JIRA_HOST_URL` when currently we only look for `ACCOUNT_USERNAME` (not `TF_ACCOUNT_USERNAME`) would've been a new usage pattern.

**Review by efabens — APPROVED** [review]

File: docs/resources/tozny_pam_jira_plugin.md (hunk 2 above)
Comment (pirtleshell) [reply docs/resources/tozny_pam_jira_plugin.md:90]: gotcha! yeah this is just some vanilla ol' terraform for setting a var automagically from the env

File: go.mod [hunk 4]
```diff
@@ -5,6 +5,6 @@ go 1.16
 require (
 	github.com/google/uuid v1.2.0
 	github.com/hashicorp/terraform-plugin-sdk/v2 v2.0.1
-	github.com/tozny/e3db-clients-go v0.0.194
+	github.com/tozny/e3db-clients-go v0.0.198-0.20211129172200-06b838d3685d
```
Comment (pirtleshell) [inline go.mod:8]: updated to current feature branch of e3db-clients-go. this should be fine so long as we don't release a new version of our provider before the client code gets checked in.

**Pull request merged**